packages/discovery-react-components/src/components/DocumentPreview/components/Highlight/passages.tsx

```diff
@@ -57,6 +57,6 @@ export function getPassagePageInfo(
 }
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function isPassage(obj: any): boolean {
+export function isPassage(obj: any): obj is QueryResultPassage {
```

Comment:
I don't understand this change. Is it correct?

Should this be this instead, and we can remove the `eslint` comment?

```js
export function isPassage(obj: QueryResultPassage): boolean {
```

Reply:
Yes, I wanted to introduce [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) for TS compiler to assume the `obj` is `QueryResultPassage` when the function returns `true`. Using this, we can avoid (error-prone) type annotations in caller.

I just realized that I haven't updated the callers... I'll do that.